src/libraries/System.Security.Cryptography/src/System/Security/Cryptography/X509Certificates/StorePal.Windows.Export.cs

```diff
@@ -102,6 +90,165 @@ public void MoveTo(X509Certificate2Collection collection)
             }
         }
 
+        public byte[] ExportPkcs12(Pkcs12ExportPbeParameters exportParameters, SafePasswordHandle password)
+        {
+            bool tryAes256sha256 = exportParameters is
+                Pkcs12ExportPbeParameters.Default or
+                Pkcs12ExportPbeParameters.Pbes2Aes256Sha256;
+
+            byte[] exported = ExportPkcs12(tryAes256sha256, password, out bool aes256Sha256);
+
+            // What we asked for and what we got are the same - so return it as-is.
+            if (tryAes256sha256 == aes256Sha256)
+            {
+                return exported;
+            }
+
+            if (!tryAes256sha256)
+            {
+                Debug.Fail("3DES PKCS12 export failed.");
+                throw new CryptographicException();
+            }
+
+            return ReEncryptAndSealPkcs12(exported, password, Helpers.WindowsAesPbe);
+        }
+
+        public byte[] ExportPkcs12(PbeParameters exportParameters, SafePasswordHandle password)
+        {
+            byte[] exported = ExportPkcs12(preferAes256Sha256: true, password, out _);
+            return ReEncryptAndSealPkcs12(exported, password, exportParameters);
```

Comment:
Is it worth checking to see if `exportParameters` happens to exactly align to the Windows AES-256/SHA-2-256 option?  I guess we'd have to do a test export once to find out what Windows' KDF counts are to see if we're aligned?

Reply:
I considered that to return it "as is" and decided not to. We would need to crack open the PKCS12 to make sure the KDF counts matched (because registry tweaks can change them). At that point we already have done the work the crack open the PKCS12, might as well just assemble it the right way. Keeps the code simpler.